File: easymall/src/main/java/org/ruikun/entity/Comment.java
package org.ruikun.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 商品评论表
 */
@Data
@TableName("comment")
public class Comment {
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 商品ID
     */
    private Long productId;

    /**
     * 订单ID
     */
    private Long orderId;

    /**
     * 评论内容
     */
    private String content;

    /**
     * 评分 1-5星
     */
    private Integer rating;

    /**
     * 评论图片，多个用逗号分隔
     */
    private String images;

    /**
     * 商家回复
     */
    private String reply;

    /**
     * 回复时间
     */
    private LocalDateTime replyTime;

    /**
     * 是否显示 0-隐藏 1-显示
     */
    private Integer showStatus;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer deleted;
}

File: easymall-frontend/src/types/comment.ts
export interface CommentPageItem {
  id: number
  userId: number
  username: string
  nickname: string
  productId: number
  productName: string
  content: string
  rating: number
  showStatus: number
  reply: string
  createTime: string
}

export interface CommentQuery {
  pageNum: number
  pageSize: number
  productId?: number
  userId?: number
  showStatus?: number
  rating?: number
}

// 用户端评论类型
export interface UserCommentVO {
  id: number
  userId: number
  userNickname: string
  userAvatar: string
  productId: number
  orderId: number
  content: string
  rating: number
  images: string
  reply: string
  replyTime: string
  createTime: string
}

export interface CommentCreateDTO {
  productId: number
  orderId: number
  content: string
  rating: number
}
